DNM: IP Address Information Fetch

zahnno opened the pull request:
Fetches IP Address Info from an IP address information provider and makes it available to the client context via the ipAddressInfo exported property.

IpAddressInfo Provider configurable via service client.

jlhughes committed:
add example p() entries for benchmark running

jlhughes committed:
add example p() entries for benchmark running

nanoNeel committed:
add benchmarkRunnerScript

nanoNeel committed:
add benchmarkRunnerScript

nanoNeel committed:
convert to use original benchmark

nanoNeel committed:
convert to use original benchmark

nanoNeel committed:
fix invalid # of benchmarks ran

nanoNeel committed:
fix invalid # of benchmarks ran

nanoNeel committed:
update to use benchmarkRunner

nanoNeel committed:
update to use benchmarkRunner

nanoNeel committed:
clean up

nanoNeel committed:
clean up

nanoNeel committed:
add benchmarkDAO

nanoNeel committed:
add benchmarkDAO

nanoNeel committed:
update benchmarkRunner to support old benchmarks

nanoNeel committed:
update benchmarkRunner to support old benchmarks

nanoNeel committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into NP-4916/BuildBenchmarkSupport

nanoNeel committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into NP-4916/BuildBenchmarkSupport

nanoNeel committed:
add support to clear PMs

nanoNeel committed:
add support to clear PMs

hchoura committed:
add benchmarkDAO
Conflicts:
	src/services

hchoura committed:
update Build benchmark support

hchoura committed:
Merge branch 'NP-4916/BuildBenchmarkSupport' of
https://github.com/kgrgreer/foam3.git into NP-4916/BuildBenchmarkSupport

Conflicts:
	src/foam/nanos/bench/BenchmarkRunner.js
	src/foam/nanos/medusa/benchmark/scripts.jrl

hchoura committed:
Merge branch 'development' into NP-4916/BuildBenchmarkSupport

nanoNeel committed:
add in changes from dev

hchoura committed:
Add benchmark result and update benchmarkRunner

hchoura committed:
Merge branch 'NP-4916/BuildBenchmarkSupport' of https://github.com/kgrgreer/foam3.git into NP-4916/BuildBenchmarkSupport

hchoura committed:
fix the ops

hchoura committed:
add mixin

hchoura committed:
add jvm info

hchoura committed:
Add JavaVmName and JavaVmVersion

hchoura committed:
fix typo

hchoura committed:
typo

Commit:
Migrated ApprovalRequest.ClassificationEnum to a ApprovalRequest.Classification reference and added an ApprovalRequestClassificationDAO

zahnno committed:
IP Address Information Fetch

Commit:
Cleanup comments

Commit:
Added some locales for ApprovalRequestClassifications

zahnno committed:
Instance condition in client info fetch

jlhughes commented on src/foam/net/ip/ClientIPAddressInfoFetch.js:
```diff
@@ -0,0 +1,35 @@
+  properties: [
+    {
+      class: 'URL',
+      name: 'ipInfoProvider',
```
@zahnno can you consider or investigate that the final real 'fetch' will fail early if one is running localhost with no network connection for example. Thanks.

jlhughes replied:
Otherwise, looks good.

zahnno replied:
I could check if the user is connected to the internet but I'm assuming this consideration is deeper than that. Got any other info?

jlhughes replied:
perhaps the client nspec configuration can be empty for foam/src so it does nothing on localhost and CI, and we can configure it when not localhost.  and then, as a developer, you can still set it up locally to test.

zahnno replied:
empty nspec - hmm, any calls to it would throw a no function error unless we abstract it to a class with an noop method.

jlhughes replied:
the url can be empty, and the fetch does nothing when undefined.

zahnno committed:
Warn user of no connection when retrieving IP Address Info

zahnno committed:
Implemented polymorphic behavior to clientIP info decorator

zahnno committed:
Added nspec entry and condition on empty provider

kgrgreer commented:
Can we review this after next standup?

minsunny committed:
daoUpdateView message

Commit:
Merge branch 'development' into NP-5130/ApprovalMigrationFOAM

Commit:
Enable authorization on approvalRequestClassificationDAO

Commit:
Change how locale is grabbed in ApprovalRequestClassification

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into NP-4916/BuildBenchmarkSupport

jlhughes committed:
Merge branch 'NP-4916/BuildBenchmarkSupport' of github.com:kgrgreer/foam3 into NP-4916/BuildBenchmarkSupport

jlhughes committed:
Continue refactor of Benchmark from Interface to Class, and BenchmarkResult centralized collection

Commit:
Merge branch 'development' into NP-5130/ApprovalMigrationFOAM

Commit:
Update src/services

Co-authored-by: Joel Hughes <jlhughes@users.noreply.github.com>

jlhughes committed:
working version

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into NP-4916/BuildBenchmarkSupport

jlhughes committed:
Rename references to Benchmark with BenchmarkRunner

jlhughes committed:
add support for running BenchmarkRunner by id, class name, or class name with 'Runner' removed - which is generally the name of the Benchmark itself.

jlhughes committed:
Move benchmarks from bench/benchmarks to benchmark/ and add journal config

jlhughes committed:
add jenkins configuration example

jlhughes committed:
Remove jenkins config - moved to nanopay

jlhughes committed:
add missing dao setup

jlhughes committed:
example config for client/server

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into NP-4916/BuildBenchmarkSupport

jlhughes committed:
add property to control log level

jlhughes committed:
Support setting RES_JAR_HOME from both environment and JVM parameters

jlhughes committed:
Support additional lookups for benchmark run requests by the benchmark name rather than the benchmark runner name

jlhughes committed:
Remove a medusa nspec from the general foam services (#967)

Nauna committed:
helper Subject.isUserInSubject

jlhughes committed:
Merge branch 'development' of github.com:kgrgreer/foam3 into NP-4916/BuildBenchmarkSupport

jlhughes committed:
Rename benchmarkResultDAOServer to benchmarkResultServerDAO

kgrgreer commented on src/foam/nanos/controller/ApplicationController.js:
```diff
@@ -346,6 +348,15 @@ foam.CLASS({
+    },
+    {
+      class: 'FObjectProperty',
+      name: 'ipAddressInfo',
```
If still needed, just make a new NSpec, instead of adding to ApplicationController.

sarthak-marwaha committed:
Move svg sizing to CSS to make it less specific
Allows for overriding where required

kgrgreer committed:
Merge pull request #969 from sarthak-marwaha/NP-5557/ButtonSVGSizing

Move button svg sizing to CSS to make it less specific

Commit:
Removed url and requesteeSession from RFITicket

Commit:
use reset(x) instead of x in OrX

jlhughes committed:
Merge pull request #970 from jinn9/SessionBugFix

Fix session caching bug

sarthak-marwaha committed:
Fix CSS ordering

Commit:
Merged dev

sarthak-marwaha committed:
Fix DAOBrowserView Actions

sarthak-marwaha committed:
Fix icon size in TitledArrayView

sarthak-marwaha committed:
Use themeGlyph in DAOUpdateView

jlhughes committed:
Correct port to 8080 - this is the true base port regardless of ssl

jlhughes committed:
Merge pull request #944 from kgrgreer/NP-5130/ApprovalMigrationFOAM

ApprovalRequest Classification Migration

Nauna committed:
Update src/foam/nanos/auth/Subject.js

minsunny committed:
updated message

kgrgreer committed:
Merge pull request #973 from kgrgreer/NP-5501/DAOUpdateViewMessage

dao update view message

minsunny committed:
updated created View

kgrgreer committed:
Merge pull request #974 from kgrgreer/NP-5501/DAOCreateViewMessage

updated DAOcreated view message

jlhughes committed:
Merge pull request #968 from kgrgreer/updateSub

helper Subject.isUserInSubject

kgrgreer committed:
Merge pull request #971 from sarthak-marwaha/NP-5557/ButtonSVGSizing

Np 5557/button svg sizing

Commit:
Readded fields, renamed requesteeSession to sessionToken

jlhughes committed:
Merge pull request #612 from kgrgreer/NP-4916/BuildBenchmarkSupport

[NP-4916] Benchmark Runner - modelled, reporting

Commit:
Updated url field to use sessionToken in expression

jlhughes committed:
Merge pull request #972 from kgrgreer/RFIOTLogin

Renamed requesteeSession to sessionToken in RFITicket

Commit:
Fix the toSummary call which shows up in detail views for ApprovalRequest

kgrgreer committed:
Merge pull request #976 from kgrgreer/NP-5597/FixApprovalRequestToSummary

Fix toSummary for ApprovalRequest

jlhughes committed:
Test if user properties have changed to invalidate the session cache

jlhughes committed:
Additional test for null subjectUser before compare

jlhughes committed:
Merge pull request #977 from kgrgreer/NP-5586/LanguageDropdown

Test if user properties have changed to invalidate the session cache

jlhughes committed:
To protect against shrc.custom not read at startup, fail 'up' if System property 'CLUSTER' not found for mediator which would otherwise report 'up'

jlhughes committed:
Merge pull request #978 from kgrgreer/NP-5504/MedusaCluster

To protect against shrc.custom not read at startup, fail 'up' if Syst…

jlhughes committed:
port medusa benchmarks to new benchmark journals

jlhughes committed:
Merge pull request #979 from kgrgreer/NP-4916/MedusaBenchmarks

port medusa benchmarks to new benchmark journals

jlhughes committed:
Status on Exception was incorrectly setting to RUNNING, rather than ERROR

jlhughes committed:
Merge pull request #980 from kgrgreer/NP-4916/BenchmarkRunnerStatus

Status on Exception was incorrectly setting to RUNNING, rather than E…

jlhughes committed:
Set minimum threads for small VMs and low core development hardware

jlhughes committed:
Replace threadpool serviceClass with serviceScript so it can be superceeded elsewhere to enabled debugging

jlhughes committed:
Merge pull request #981 from kgrgreer/AbstractFixedThreadPoolMin

Abstract fixed thread pool min

jlhughes committed:
Throw MedusaException during Replaying

On mediator restart, existing admin logins will attempt http operations,
resulting in SessionServerBox 'put' operations on the sessionDAO,
which in turn cause 'put' from the secondaries to the primary.
But during replay, either the primary is not found, or the request
is sent, and the calling thread blocks in ReplayingDAO.  The browser
sends multiple requests and can potentially block a large number of
threads, causing resource issues for replaying.

jlhughes committed:
Merge pull request #982 from kgrgreer/NP-5125/MedusaAdapterReplaying

Throw MedusaException during Replaying

zahnno committed:
Added readonly service for ip address infos

zahnno committed:
Merge branch 'development' of https://github.com/kgrgreer/foam3 into ip-address-info

zahnno committed:
Fetch IP Address Info DAO
